sdk/testing/stepwise/stepwise.go

```diff
@@ -0,0 +1,351 @@
+	if !testing.Verbose() {
+		tt.Fatal("Acceptance tests must be run with the -v flag on tests")
+		return
+	}
+
+	// Run the PreCheck if we have it
+	if c.PreCheck != nil {
+		c.PreCheck()
+	}
+
+	// Create an in-memory Vault core
+	// TODO use test logger if available?
+	logger := logging.NewVaultLogger(log.Trace)
+	if c.Environment == nil {
+		tt.Fatal("nil driver in acceptance test")
+		// return here only used during testing when using mockT type, otherwise
+		// Fatal will exit
+		return
+	}
+
+	err := c.Environment.Setup()
+	if err != nil {
+		driverErr := fmt.Errorf("error setting up driver: %w", err)
+		if !c.SkipTeardown {
+			if err := c.Environment.Teardown(); err != nil {
+				driverErr = fmt.Errorf("error during driver teardown: %w", driverErr)
+			}
+		}
+		tt.Fatal(driverErr)
+	}
+
+	defer func() {
+		if c.SkipTeardown {
+			logger.Info("driver Teardown skipped")
+			return
+		}
+		if err := c.Environment.Teardown(); err != nil {
+			logger.Error("error in driver teardown:", "error", err)
+		}
+	}()
+
+	// retrieve the root client from the Environment. If this returns an error, fail
+	// immediately
+	rootClient, err := c.Environment.Client()
+	if err != nil {
+		tt.Fatal(err)
+	}
+
+	// Trap the rootToken so that we can preform revocation or other tasks in the
+	// event any steps remove the token during testing.
+	rootToken := c.Environment.RootToken()
+
+	// Defer revocation of any secrets created. We intentionally enclose the
+	// responses slice so in the event of a fatal error during test evaluation, we
+	// are still able to revoke any leases/secrets created
+	var responses []*api.Secret
+	defer func() {
+		// restore root token for admin tasks
+		rootClient.SetToken(rootToken)
+		// failedRevokes tracks any errors we get when attempting to revoke a lease
+		// to log to users at the end of the test.
+		var failedRevokes []*api.Secret
+		for _, secret := range responses {
+			if secret.LeaseID == "" {
+				continue
+			}
+
+			if err := rootClient.Sys().Revoke(secret.LeaseID); err != nil {
+				tt.Error(fmt.Errorf("error revoking lease: %w", err))
+				failedRevokes = append(failedRevokes, secret)
+				continue
+			}
+		}
+
+		// If we have any failed revokes, log it.
+		if len(failedRevokes) > 0 {
+			for _, s := range failedRevokes {
+				tt.Error(fmt.Sprintf(
+					"WARNING: Revoking the following secret failed. It may\n"+
+						"still exist. Please verify:\n\n%#v",
+					s))
+			}
+		}
+	}()
+
+	stepCount := len(c.Steps)
+	for i, step := range c.Steps {
+		// range is zero based, so add 1 for a human friendly output of steps.
+		// "index" here is only used for logging / output, and not to reference the
+		// step in the slice of steps.
+		index := i + 1
+		if logger.IsWarn() {
+			progress := fmt.Sprintf("%d/%d", index, stepCount)
+			logger.Warn("Executing test step", "step_number", progress)
+		}
+
+		// reset token in case it was cleared
+		client, err := rootClient.Clone()
```

Comment:
This client isn't used beyond `SetToken` below - should it be passed into `makeRequest` instead of the environment?

Reply:
I'm going to defer addressing this for now. In the future I would like to support each test case or test step being ran with a token created with user defined policies attached instead of using the root token. At this time I don't know what that looks like, so for now I would like to leave this as is to continue to support the `Unauthenticated` feature and revisit this when I have a better idea on that feature.